lib/services/dialog_import_service.rb

```diff
@@ -106,23 +106,17 @@ def import_from_dialogs(dialogs)
     raise ParsedNonDialogYamlError if dialogs.empty?
     associations = []
     dialogs.each do |dialog|
-      dialog["dialog_tabs"].flat_map do |tab|
-        tab["dialog_groups"].flat_map do |group|
-          group["dialog_fields"].flat_map do |field|
-            associations << { field["name"] => field["dialog_field_responders"] } unless field["dialog_field_responders"].nil?
-          end
-        end
-      end
       new_or_existing_dialog = Dialog.where(:label => dialog["label"]).first_or_create
       dialog['id'] = new_or_existing_dialog.id
+      associations_to_be_created = build_association_list(dialog)
       new_or_existing_dialog.update_attributes(
         dialog.merge(
           "dialog_tabs"      => build_dialog_tabs(dialog),
           "resource_actions" => build_resource_actions(dialog)
         )
       )
       fields = new_or_existing_dialog.dialog_fields
-      associations.each do |association|
+      associations_to_be_created.each do |association|
```

Comment:
If we're not using `associations_to_be_created` anywhere else can we just call `.each` directly on `build_association_list(dialog)`?

If you'd rather have the variable that's fine, but can we move it to line 118 instead of 111 so that it's not separated by unrelated code?

Reply:
Have to have it before the update_attributes call, I think